pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -421,7 +423,7 @@ func (r *controlPlaneInstanceReconciler) PostStatus(ctx context.Context) error {
 	log.Info("Posting status update", "conditions", r.Status.Conditions)
 	if err := r.Client.Get(ctx, client.ObjectKey{Name: r.Instance.Name, Namespace: r.Instance.Namespace}, instance); err == nil {
 		instance.Status = *r.Status.DeepCopy()
-		if err = r.Client.Status().Patch(ctx, instance, common.NewStatusPatch(instance.Status)); err != nil && !(apierrors.IsGone(err) || apierrors.IsNotFound(err)) {
+		if err = r.Client.Status().Update(ctx, instance); err != nil && !(apierrors.IsGone(err) || apierrors.IsNotFound(err)) {
```

Comment:
This will have a significant effect on the entire operation of the reconciler, because the UPDATE operation will fail on conflicts, whereas PATCH won't. A failed update causes the reconciler to imediately run again, at which point it may fail again until the local cache is finally updated by the watch event.

Reply:
I fixed how we perform the patch operation in https://github.com/maistra/istio-operator/pull/606. Please review the PR and then revert this change.

Reply:
done.  i was worried about that.  thanks for fixing it correctly.